subprojects/reporting/src/main/java/org/gradle/api/reporting/ReportingExtension.java

```diff
@@ -54,13 +55,13 @@ public abstract class ReportingExtension {
 
     private final ProjectInternal project;
     private final DirectoryProperty baseDirectory;
-    private final ExtensiblePolymorphicDomainObjectContainer<ReportSpec> reports;
+    private final ReportSpecContainer reports;
 
     @Inject
-    public ReportingExtension(Project project) {
+    public ReportingExtension(Project project, InstantiatorFactory instantiatorFactory, ServiceRegistry servicesToInject, CollectionCallbackActionDecorator collectionCallbackActionDecorator) {
```

Comment:
:x: this is a binary breaking change in public API

You should keep the original constructor without `@Inject` and deprecate it.

Reply:
Hmm... any idea why our binary compatibility tests aren't picking it up?

Reply:
Ah, if the binary check doesn't pick it we probably have a rule for injectable ctors now. Ignore me :)